include/deal.II/numerics/vector_tools.templates.h

```diff
@@ -6321,12 +6326,10 @@ namespace VectorTools
                                   "with at least 'dim' components. In that case, this function "
                                   "will take the divergence of the first 'dim' components."));
               double sum = 0;
-              // take the trace of the derivatives, square it, multiply it
-              // with the weight function
+              // take the trace of the derivatives scaled by the weight and square it
               for (unsigned int k=0; k<dim; ++k)
-                sum += (data.psi_grads[q][k][k] * data.psi_grads[q][k][k]) *
-                       data.weight_vectors[q](k);
-              diff += sum * fe_values.JxW(q);
+                sum += data.psi_grads[q][k][k] * std::sqrt(data.weight_vectors[q](k));
+              diff += sum * sum * fe_values.JxW(q);
```

Comment:
In other words, it's ok with me if you want to leave it that way. In that case, please merge.

The alternative would be to assert that the weight is scalar, and then just write it as
```
  for (k=...)
    sum += data.psi_grads[q][k][k];
  diff += sum * sum * weight[0] * JxW(q);
```
i.e., to explicitly use the zero-th weight.